sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/SearchIndexAsyncClient.java

```
@@ -0,0 +1,369 @@
+import static com.azure.core.util.FluxUtil.pagedFluxError;
+import static com.azure.core.util.FluxUtil.withContext;
+
+/**
+ * Asynchronous Client to manage and query search index, as well as manage other resources,
+ * on a Cognitive Search service.
+ */
+public class SearchIndexAsyncClient {
+    /**
+     * The logger to be used
+     */
+    private final ClientLogger logger = new ClientLogger(SearchIndexAsyncClient.class);
+
+    /**
+     * The underlying AutoRest client used to interact with the Search service
+     */
+    private final SearchServiceRestClientImpl restClient;
+
+    SearchIndexAsyncClient(String endpoint, SearchServiceVersion serviceVersion, HttpPipeline httpPipeline) {
+        this.restClient = new SearchServiceRestClientBuilder()
+            .endpoint(endpoint)
+            .apiVersion(serviceVersion.getVersion())
+            .pipeline(httpPipeline)
+            .build();
+    }
+
+    /**
+     * Creates a new Azure Cognitive Search index.
+     *
+     * @param index definition of the index to create.
+     * @return the created Index.
+     */
+    public Mono<SearchIndex> create(SearchIndex index) {
+        return createWithResponse(index, null).map(Response::getValue);
+    }
+
+    /**
+     * Creates a new Azure Cognitive Search index.
+     *
+     * @param index definition of the index to create
+     * @param requestOptions additional parameters for the operation. Contains the tracking ID sent with the request to
+     * help with debugging
+     * @return a response containing the created Index.
+     */
+    public Mono<Response<SearchIndex>> createWithResponse(SearchIndex index, RequestOptions requestOptions) {
+        return withContext(context -> createWithResponse(index, requestOptions, context));
+    }
+
+    Mono<Response<SearchIndex>> createWithResponse(SearchIndex index, RequestOptions requestOptions,
+        Context context) {
+        Objects.requireNonNull(index, "'Index' cannot be null");
```

We should continue this with monoError(logger, new NullPointerException("'index' cannot be null."); This will throw rhater than return that mono error we want.

---

Good catch. Will rework on all occurrences.

---

Or just include it within the `try {} catch()` so that it is raised as a monoError

---

Not sure if this applies to Java, but try/catch is expensive in .NET where we can just throw the exception as-is.